pkg/api/types.go

```diff
@@ -204,6 +204,8 @@ type VolumeSource struct {
 
 // Similar to VolumeSource but meant for the administrator who creates PVs.
 // Exactly one of its members must be set.
+// All PersistentVolumes are volumes, only some Volumes are persistent.  All pointers
```

Comment:
You need to distinguish the implementation from the API.

In the API, PVSource and VSource will be partially overlapping sets.  In the implementation PLugin is a strict subset of VPlugin (or else you could not SetUp).  For example, we need AWSEBS or GCEPD to be a VolumePlugin, because we want to use it as a PersistentVolume, but we don't want to offer it directly as a VolumeSource (in the future).